src/Router.ts

```diff
@@ -439,8 +439,19 @@ export class Router<C extends Context> extends Evented implements RouterInterfac
 		return this._outletContextMap.has(outletId);
 	}
 
-	getOutlet(outletId: string): OutletContext | undefined {
-		return this._outletContextMap.get(outletId);
+	getOutlet(outletId: string | Array<String>): OutletContext | undefined {
+		if (Array.isArray(outletId)) {
+			const outletContextArray = arrayFrom(this._outletContextMap);
+			const outlets = arrayFrom(outletId);
+
+			const matchingOutlet = find(outletContextArray, ([key]) => {
```

Comment:
Would be nicer if you destructured this:

```ts
const [, matchingOutlet ] = find(...);
```

Then just return `matchingOutlet`

Reply:
Ah yes tried that too! TypeScript was complaining about something like undefined not being an array type and I think it makes sense. The find function can return undefined, and we can't destructure undefined. Is that your understanding too?

Reply:
If you are convinced that `find()` will not return `undefined` because of other checks or validations, you can use the _escape hatch_ of the non-null assertion operator (`find(...)!`).  Our general convention is to comment the code though why you believe you are safe using the operator.

Reply:
Oh cool thanks for explaining that! In this particular case, I'm not convinced that `find()` will never return `undefined` so I think we can leave it as it is, useful to know though!

Reply:
We really try and avoid using `!` to force TS to ignore `strictNullChecks` in src code.

Could `|| []` after the find if it helps make the code clearer by enabling the destructuring.

Reply:
I'm not sure the code becomes any clearer at that point. I tried it anyway and got:

![image](https://user-images.githubusercontent.com/381495/32102359-d60de452-bb13-11e7-9149-eef7feb27f2b.png)

Don't suppose you know anything about that?